sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -3662,3 +3682,52 @@ class SimpleDelegatingCatalog extends DelegatingCatalogExtension {
     super.createTable(ident, columns, partitions, newProps)
   }
 }
+
+
+class V2CatalogSupportBuiltinDataSource extends DelegatingCatalogExtension {
```

Comment:
can we test with an individual catalog that doesn't extend `DelegatingCatalogExtension`?

Reply:
oh It's registered at https://github.com/apache/spark/pull/47723/files#diff-0c0de83250682f28a4d129e4b1c0290b2725dcaa36a5eee51ea922f0d53a9a43R44

Reply:
Ok then I just choose to extend `InMemoryCatalog` and test.